### ISSUE TITLE
`Hex dependency resolution failed` when installing alongside Phoenix

### ISSUE THREAD

**Issue by arcanemachine** [ash-project/ash_postgres#134]
**Describe the bug**
When following the `Add Dependencies` section of the `ash_phoenix` "Getting Started" guide, the required dependencies cannot be installed without error. Although this is a Phoenix/ash_phoenix issue, the lock file complains about the dependencies in `ash_postgres` so I posted the issue here.

**To Reproduce**

These versions are the results of the suggested instructions to "use `mix hex.info <library_name>` to get the latest versions of each dependency":

```
{:phoenix, "~> 1.7.2"},                                                                                                                                                                                                   
...
{:ash, "~> 2.6"},                                                                                                                                                                                                   
{:ash_postgres, "~> 1.3"},                                                                                                                                                                                          
{:ash_phoenix, "~> 1.2"}                                                                                                                                                                                            
```

Attempting to run `mix deps.get` produces the following result:

```
Resolving Hex dependencies...
Resolution completed in 0.266s
Because "the lock" specifies nimble_options 1.0.1 and ash_postgres >= 1.2.0-rc.0 depends on nimble_options ~> 0.5, the lock is incompatible with ash_postgres >= 1.2.0-rc.0.
And because your app depends on the lock, ash_postgres >= 1.2.0-rc.0 is forbidden.
So, because your app depends on ash_postgres ~> 1.3, version solving failed.
** (Mix) Hex dependency resolution failed
```

Instructions given above.

**Expected behavior**
The dependencies should be installed.

** Runtime
 - Elixir version - 1.14.3
 - Erlang version - Erlang/OTP 25 [erts-13.2]
 - OS - Xubuntu 22.04.2 LTS
 - Ash version - 2.6
 - any related extension versions - `phoenix` ~> 1.7.2, `ash_postgres` ~> 1.3, `ash_phoenix` ~> 1.2

**Additional context**
Setting `{nimble_options, "~> 0.5"}` allows the dependencies to be installed, but I don't know if this causes any problems with Phoenix.

**Comment by zachdaniel** [comment]
Interesting...I think Ash packages can be on nimble options 1.0+. I'll look at updating it.

**Comment by zachdaniel** [comment]
Once CI passes, a new version of spark will release that should solve for this problem. So you can do `mix deps.update spark` and it should use nimble options 1.0

**Comment by zachdaniel** [comment]
Closing this for now unless there is still an issue after updating spark :)

**Comment by arcanemachine** [comment]
First of all, thanks for the incredibly quick response.

However, there is still an explicit dependency on `{:nimble_options, "~> 0.5"}` in the `ash_postgres` repo, which causes the same error message to appear when doing the `ash_phoenix` tutorial.

I was able to get a working copy locally using the following steps (I'm an Elixir noob if it wasn't obvious):

- Make a clone of this repo
- Update the dependency to `{:nimble_options, "~> 1.0"},`
- Run `mix deps.update spark`
- Then I could build a release with `mix release`
- Then I installed the deps from my "hello world" ash_phoenix project using the local `ash_postgres` dependency that I built, and everything appears to be working (so far...)

So I believe that the explicit `nimble_options` dependency in `ash_postgres` itself still needs to be updated to fully fix the issue. (Or perhaps it could be removed altogether? I tried compiling `ash_postgres` with the `nimble_options` commented out, and it appeared to complete without error.)

Thanks again.
